### PR TITLE
Homework for first lecture.

### PR DESCRIPTION
**Pull request by Khazel**
Completed homework tasks for the first lecture.

### CODE REVIEW

File: src/Lecture_1/Homework/Task_1.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        System.out.println("Please enter A: ");
+        double a = sc.nextDouble();
+
+        System.out.println("Please enter B: ");
+        double b = sc.nextDouble();
+
+        System.out.println("Please enter C: ");
+        double c = sc.nextDouble();
+
+        if ((c > a) && (c < b)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_1.java]: You are missing second case whre B < C < A
(C > B && C < A)

File: src/Lecture_1/Homework/Task_2.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        int firstInteger = sc.nextInt();
+
+        System.out.println("Please enter second integer: ");
+        int secondInteger = sc.nextInt();
+
+        int integerSum = firstInteger + secondInteger;
+        int integerDifference = firstInteger - secondInteger;
+        int integerProduct = firstInteger * secondInteger;
+        int integerReminderOfDivision = firstInteger % secondInteger;
+        int integerDivision = firstInteger / secondInteger;
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_2.java]: What will happened id second integer = 0 ?
You should validate the user input.

File: src/Lecture_1/Homework/Task_2.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        double firstFloatingPointNumber = sc.nextDouble();
+
+        System.out.println("Please enter second floating point number: ");
+        double secondFloatingPointNumber = sc.nextDouble();
+
+        double floatingPointNumberSum = firstFloatingPointNumber + secondFloatingPointNumber;
+        double floatingPointNumberDifference = firstFloatingPointNumber - secondFloatingPointNumber;
+        double floatingPointNumberProduct = firstFloatingPointNumber * secondFloatingPointNumber;
+        double floatingPointNumberRemainderOfDivision = firstFloatingPointNumber % secondFloatingPointNumber;
+        double floatingPointDivision = firstFloatingPointNumber / secondFloatingPointNumber;
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_2.java]: same... secondFloatingPointNumber = 0 ?

File: src/Lecture_1/Homework/Task_5.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import java.util.Scanner;
+
+public class Task_5 {
+
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Please inform if you are at work:");
+        String atWork = sc.nextLine();
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_5.java]: user must provide you a boolean.
"... am I at work (boolean type)."

File: src/Lecture_1/Homework/Task_5.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        boolean isAtWork;
+        if (atWork == "Yes") isAtWork = true;
+        else {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_5.java]: Could be like:
boolean isAtWork = (atWork == "Yes") ? true : false;

File: src/Lecture_1/Homework/Task_6.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        int fourthDigit = number%10;
+
+        int firstNumber = firstDigit * 10 + fourthDigit;
+        int secondNumber = secondDigit * 10 + thirdDigit;
+
+        System.out.println("Output:");
+        if (firstNumber > secondNumber) {
+            System.out.println(firstNumber + ">" + secondNumber);
+        } else {
+            System.out.println(firstNumber + "<" + secondNumber);
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_6.java]: What will be the output when I enter 1111?

File: src/Lecture_1/Homework/Task_6.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import java.util.Scanner;
+
+public class Task_6 {
+
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Please enter 1 four-digit natural number from the keyboard that falls within the range [1000 – 9999].");
+        int number = sc.nextInt();
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_6.java]: What will happened if I enter 456?
Validate user input! number must be [1000 – 9999].

File: src/Lecture_1/Homework/Task_7.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        int secondNumber = sc.nextInt();
+
+        System.out.println("Input:");
+        System.out.println(firstNumber + ", " + secondNumber);
+
+        int multiplicationResult = firstNumber * secondNumber;
+
+        System.out.println("Output:");
+        if (multiplicationResult % 2 == 0)
+        {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_7.java]: The formating is wrong. "{" must be on the same line as the "if" statement.

File: src/Lecture_1/Homework/Task_8.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import java.util.Scanner;
+
+public class Task_8 {
+
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Please enter a three-digit number from the keyboard, without the digit 0 in this number");
+        int number = sc.nextInt();
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_8.java]: Validation!

File: src/Lecture_1/Homework/Task_8.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+        int firstDigit = number/100%10;
+        int secondDigit = number/10%10;
+        int thirdDigit = number%10;
+
+        if (number % firstDigit == 0)
+        {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_8.java]: The formating is wrong. "{" must be on the same line as the "if" statement.

File: src/Lecture_1/Homework/Task_9.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+        System.out.println("Please enter an integer in the range [0-100]");
+
+        Scanner sc = new Scanner(System.in);
+        int integer = sc.nextInt();
+        // It does not work properly on the border values 5, 20, 50, 80, 100 as task does not mention what to do with them.
+        if (integer == 0) {
+            System.out.println("0 – it will not rain");
+        }
+        if ((integer > 0) && (integer < 5)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_9.java]: if (integer <= 5) - first part is not needed it was checked above and don't forghet that 5 must be included in this case

File: src/Lecture_1/Homework/Task_9.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+            System.out.println("Between 0 and 5 - minimal chance of precipitation");
+        }
+        if ((integer > 5) && (integer < 20)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_9.java]: first part is not needed it was checked above and don't forghet that 20 must be included in this case

File: src/Lecture_1/Homework/Task_9.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+            System.out.println("Between 5 and 20 - a small chance of precipitation");
+        }
+        if ((integer > 20) && (integer < 50)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_9.java]: first part is not needed it was checked above and don't forghet that 50 must be included in this case

File: src/Lecture_1/Homework/Task_9.java [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+            System.out.println("Between 20 and 50 - there is a chance of precipitation");
+        }
+        if ((integer > 50) && (integer < 80)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_9.java]: same...

File: src/Lecture_1/Homework/Task_9.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+            System.out.println("Between 50 and 80 - high chance of precipitation");
+        }
+        if ((integer > 80) && (integer < 100)) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_1/Homework/Task_9.java]: same...

File: src/Lecture_1/Homework/Task_1.java (hunk 1 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_1.java]: Added validation for the case when B < C < A.

File: src/Lecture_1/Homework/Task_2.java (hunk 2 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_2.java]: integerReminderOfDivision, integerDivision operations will fail.

Added validation for zero in user input.

File: src/Lecture_1/Homework/Task_2.java (hunk 3 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_2.java]: floatingPointNumberReminderOfDivision, floatingPointNumberDivision operations will fail.

Added validation for zero in user input.

File: src/Lecture_1/Homework/Task_5.java (hunk 4 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_5.java]: Updated variable to accept user input as true or false.

File: src/Lecture_1/Homework/Task_5.java (hunk 5 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_5.java]: Updated variable to accept user input as true or false.

File: src/Lecture_1/Homework/Task_6.java (hunk 6 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_6.java]: Added validation for case when all four numbers are equal.

File: src/Lecture_1/Homework/Task_6.java (hunk 7 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_6.java]: Added validation for acceptable user input range.

File: src/Lecture_1/Homework/Task_7.java (hunk 8 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_7.java]: Corrected the formatting.

File: src/Lecture_1/Homework/Task_8.java (hunk 9 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_8.java]: Added validation for user input to not contain zeroes.

Added validation for acceptable user input range.

File: src/Lecture_1/Homework/Task_8.java (hunk 10 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_8.java]: Corrected the formatting.

File: src/Lecture_1/Homework/Task_9.java (hunk 11 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_9.java]: Reformatted conditions.

File: src/Lecture_1/Homework/Task_9.java (hunk 12 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_9.java]: Reformatted conditions.

File: src/Lecture_1/Homework/Task_9.java (hunk 13 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_9.java]: Reformatted conditions.

File: src/Lecture_1/Homework/Task_9.java (hunk 14 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_9.java]: Reformatted conditions.

File: src/Lecture_1/Homework/Task_9.java (hunk 15 above)
Comment (Khazel) [reply src/Lecture_1/Homework/Task_9.java]: Reformatted conditions.

**Review by KaloyanTsvetkov — APPROVED** [review]

**Pull request merged**